pkg/cluster/bootstrap.go

```diff
@@ -29,29 +34,156 @@ func (c *Cluster) Bootstrap(ctx context.Context) error {
 		return err
 	}
 
-	shouldBootstrap, err := c.shouldBootstrapLoad(ctx)
+	shouldBootstrap, isInitialized, err := c.shouldBootstrapLoad(ctx)
 	if err != nil {
 		return err
 	}
 	c.shouldBootstrap = shouldBootstrap
 
+	if c.managedDB != nil {
+		// In the case of etcd, if the database has been initialized, it doesn't
+		// need to be bootstrapped however we still need to check the database
+		// and reconcile the bootstrap data. Below we're starting a temporary
+		// instance of etcd in the event that etcd certificates are unavailable,
+		// reading the data, and comparing that to the data on disk, all the while
+		// starting normal etcd.
+		isHTTP := c.config.JoinURL != "" && c.config.Token != ""
+		if isInitialized && !isHTTP {
+			tmpDataDir := filepath.Join(c.config.DataDir, "db", "tmp-etcd")
+			os.RemoveAll(tmpDataDir)
+			if err := os.Mkdir(tmpDataDir, 0700); err != nil {
+				return err
+			}
+			etcdDataDir := etcd.DBDir(c.config)
+			if err := createTmpDataDir(etcdDataDir, tmpDataDir); err != nil {
+				return err
+			}
+
+			args := executor.ETCDConfig{
+				DataDir:           tmpDataDir,
+				ForceNewCluster:   true,
+				ListenClientURLs:  "http://127.0.0.1:2379",
```

Comment:
we may want to change the p ort here to something else than 2379 just not to make a conflict with rke2 static pods when integrated

Reply:
Hmm, that's good thinking ahead. Do we need to ensure that the static pod etcd is dead before doing this? What happens if you copy the etcd db out from under a running etcd server? Is it going to fail due to an improperly closed DB?

Reply:
well, this is the same concern we had earlier about not closing etcd each time we restart rke2, I dont think etcd will fail when you copy the data dir, but I am also concerned about the wal file corruption, but even with that, this will only corrupt the copied temp etcd not the original one

Reply:
And I think it'd only get corrupt if there were an attempted write to that wal file.